### ISSUE TITLE
Grouped search results header has invalid markup, needs dl wrapper

### ISSUE THREAD

**Issue by seanaery** [projectblacklight/arclight#1430]
In the [index_metadata_field_component.html.erb](https://github.com/projectblacklight/arclight/blob/main/app/components/arclight/index_metadata_field_component.html.erb) view there are `<dt>`/`<dd>` pairs getting rendered without a wrapping `<dl>` element. This is invalid HTML and is getting flagged as an accessibility issue by axe DevTools.

<img width="1142" alt="Screenshot 2023-12-04 at 10 49 09 AM" src="https://github.com/projectblacklight/arclight/assets/3933756/9095427c-dbd5-4e14-a426-3ee5a4da9fb0">

More:
https://dequeuniversity.com/rules/axe/4.8/dlitem?application=AxeChrome

**Comment by lfarrell** [comment]
Fix PR: https://github.com/projectblacklight/arclight/pull/1470
